pkg/providers/k8s/k8s.go

```diff
@@ -0,0 +1,147 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/helmfile/vals/pkg/api"
+	"github.com/helmfile/vals/pkg/log"
+)
+
+type provider struct {
+	log            *log.Logger
+	KubeConfigPath string
+	KubeContext    string
+}
+
+func New(l *log.Logger, cfg api.StaticConfig) *provider {
+	p := &provider{
+		log: l,
+	}
+
+	kubeConfig, err := getKubeConfig(cfg)
+	if err != nil {
+		fmt.Printf("An error occurred getting the Kubeconfig path: %s\n", err)
+		return p
```

Comment:
what will happen after return p without the right args?

Reply:
I handled this at GetString, by returning an error on an empty param. But figured out how to handle it in New. I pushed it in the latest commit.